src/docfx/lib/markdown/validation/DocsValidationExtension.cs

```diff
@@ -88,9 +50,108 @@ internal static class DocsValidationExtension
+            {
+                documentNode = new Heading
+                {
+                    Level = headingBlock.Level,
+                    SourceInfo = headingBlock.GetSourceInfo(),
+                    ParentSourceInfoList = context.Parents?.Cast<object?>().ToList() ?? new List<object?>(),
+                    Content = GetHeadingContent(headingBlock), // used for reporting
+                    HeadingChar = headingBlock.HeaderChar,
+                    RenderedPlainText = markdownEngine.ToPlainText(headingBlock), // used for validation
+                    IsVisible = MarkdigUtility.IsVisible(headingBlock),
+                    IsCanonicalVersion = isCanonicalVersion,
+                    Zone = context.ZoneStack.TryPeek(out var z) ? z : null,
+                    Monikers = context.ZoneMoniker.ToList(),
+                };
+            }
+            else if (node is LeafBlock leafBlock)
+            {
+                documentNode = new ContentNode
+                {
+                    SourceInfo = node.GetSourceInfo(),
+                    IsVisible = MarkdigUtility.IsVisible(leafBlock),
+                    IsCanonicalVersion = isCanonicalVersion,
+                    Zone = context.ZoneStack.TryPeek(out var z) ? z : null,
+                    Monikers = context.ZoneMoniker.ToList(),
+                };
+            }
+
+            if (documentNode != null)
+            {
+                documentNodes.Add(documentNode);
+                if (context.IsInclude)
+                {
+                    if (!inclusionDocumentNodes.TryGetValue(context.Document, out var inclusionNodes))
+                    {
+                        inclusionDocumentNodes[context.Document] = inclusionNodes = new List<ContentNode>();
+                    }
+
+                    inclusionNodes.Add(documentNode);
+                }
+            }
+        }
+
+        private static void CodeBlockValidation(
+            MarkdownObject node,
+            MarkdownVisitContext context,
+            List<CodeBlockItem> codeBlockItemList,
+            string? canonicalVersion,
+            MonikerList fileLevelMoniker)
+        {
+            var isCanonicalVersion = IsCanonicalVersion(canonicalVersion, fileLevelMoniker, context.ZoneMoniker);
+
+            CodeBlockItem? codeBlockItem = null;
+
+            if (node is FencedCodeBlock fencedCodeBlock)
+            {
+                codeBlockItem = new CodeBlockItem
+                {
+                    Type = CodeBlockTypeEnum.FencedCodeBlock,
+                    Info = fencedCodeBlock.Info,
+                    Arguments = fencedCodeBlock.Arguments,
+                    IsOpen = fencedCodeBlock.IsOpen,
+                    SourceInfo = fencedCodeBlock.GetSourceInfo(),
+                    IsCanonicalVersion = isCanonicalVersion,
+                };
+            }
+            else if (node is CodeBlock codeBlock && context.TripleColonStack.Count == 0)
```

Comment:
Are there more context around this check `context.TripleColonStack.Count == 0`?

Triple colon is a broad concept, it is a general way of doing any markdown extension and is really a syntax helper.

Reply:
For now, there are no more check. We do this check in order to exclude the Indented CodeBlock which is in Triple Colon.

Reply:
This check is here to workaround the problem that triple colon block cannot tell apart leaf block vs container block. The fix on triple colon is tracked here: https://ceapex.visualstudio.com/Engineering/_workitems/edit/279439

Reply:
OK.